### PR TITLE
feat(ME-1629): support for aws ec2 connect

### PR DESCRIPTION
**Pull request by btoonk**
# Description

Support for aws EC2 Connect upstream method
Fixes # ME-1629

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested locally

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -85,7 +85,9 @@ var (
 	awsECSServices          []string
 	awsECSTasks             []string
 	awsECSContainers        []string
+	awsEC2AZ                string
```
Comment (adrianosela) [inline cmd/root.go:88]: s/awsEC2AZ/awsAvailabilityZone/g

AZs arent specific to EC2, also favour more explicit variable name

File: internal/ssh/proxy.go [hunk 2]
```diff
@@ -607,6 +630,68 @@ func handleSSHclient(conn net.Conn, config ProxyConfig) {
...
+	sshConn, chans, reqs, err := ssh.NewServerConn(conn, config.sshServerConfig)
+	if err != nil {
+		fmt.Printf("sshauthproxy: failed to accept ssh connection: %s\n", err)
+		return
+	}
+
+	user := sshConn.User()
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		log.Fatal(err)
```
Comment (adrianosela) [inline internal/ssh/proxy.go:645]: Avoid Fatal in favour of propagating error

**Review by adrianosela — CHANGES_REQUESTED** [review]
Overall LGTM. Minor changes requested.

File: internal/ssh/proxy.go [hunk 3]
```diff
@@ -607,6 +630,68 @@ func handleSSHclient(conn net.Conn, config ProxyConfig) {
...
+	defer conn.Close()
+
+	sshConn, chans, reqs, err := ssh.NewServerConn(conn, config.sshServerConfig)
+	if err != nil {
+		fmt.Printf("sshauthproxy: failed to accept ssh connection: %s\n", err)
+		return
+	}
+
+	user := sshConn.User()
+	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
```
Comment (adrianosela) [inline internal/ssh/proxy.go:643]: Key generation and PEM encoding can be moved to a separate reusable function.

File: cmd/root.go (hunk 1 above)
Comment (btoonk) [reply cmd/root.go]: ack, changed to awAvailabilityZone

File: internal/ssh/proxy.go (hunk 2 above)
Comment (btoonk) [reply internal/ssh/proxy.go]: changed.

File: internal/ssh/proxy.go (hunk 3 above)
Comment (btoonk) [reply internal/ssh/proxy.go:652]: for now we leave it here, we can always change this.

Comment (btoonk) [reply internal/ssh/proxy.go:652]: for now we leave it here, we can always change this.

**Review by adrianosela — APPROVED** [review]

**Pull request merged**